### PR TITLE
GH-33 & GH-34 Add immutable, and reselect

### PR DESCRIPTION
**Pull request by born2bewild**
Resolves GH-33, and resolves GH-34

### CODE REVIEW

File: src/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ArronaxImmutableState,
+  ArronaxState,
+  ImmutableMap
+} from './types';
+
+export const selectFiltersDomain = (
+  state: ArronaxImmutableState
+): ImmutableMap<TezosFilter> => state.get('filters');
+
+export const makeSelectFilters = createSelector<
```
Comment (vishakh) [inline src/selectors.ts]: According to the `reselect` tutorial, selectors are named using the convention `xSelector`. How how come you are using a different naming convention?

File: src/types.tsx [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
+import * as Immutable from 'immutable';
 
 export interface ArronaxState {
     filters: TezosFilter;
     dataView: DataView;
     network: string;
 }
 
-export enum DataView {Blocks, Accounts, Operations}
+export interface ImmutableState {
```
Comment (vishakh) [inline src/types.tsx]: For my understanding, can you describe a little bit how you are setting up the immutable Redux state?

File: src/components/Arronax.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ export interface ArronaxProps {
     setNetwork: (network: string) => void;
 }
 
-export const Arronax = (props: ArronaxProps) =>
+export const Arronax = (props: ArronaxProps): JSX.Element =>
```
[the remark this answers is not in the document]
Comment (born2bewild) [reply src/components/Arronax.tsx]: Eslint requirement, return type have to be defined

File: src/types.tsx (hunk 2 above)
Comment (born2bewild) [reply src/types.tsx]:
```const initialState = fromJS({
  filters: emptyFilters,	  filters: emptyFilters,
  dataView: DataView.Blocks,	  dataView: DataView.Blocks,
  network: 'zeronet'	  network: 'zeronet'
};	});```

and then on redux 
```state.set('filters', fromJS(action.filters))```

File: src/selectors.ts (hunk 1 above)
Comment (born2bewild) [reply src/selectors.ts]: previously that was the most common naming convention, should I change it ? ;-)

**Pull request closed without merging**